src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props

```diff
@@ -39,10 +39,15 @@ Copyright (c) .NET Foundation. All rights reserved.
     <FileAlignment Condition=" '$(FileAlignment)' == '' ">512</FileAlignment>
     <ErrorReport Condition=" '$(ErrorReport)' == '' ">prompt</ErrorReport>
     <AssemblyName Condition=" '$(AssemblyName)' == '' ">$(MSBuildProjectName)</AssemblyName>
-    <RootNamespace Condition=" '$(RootNamespace)' == '' ">$(MSBuildProjectName.Replace(" ", "_"))</RootNamespace>
     <Deterministic Condition=" '$(Deterministic)' == '' ">true</Deterministic>
   </PropertyGroup>
 
+  <PropertyGroup Condition=" '$(RootNamespace)' == '' ">
+    <!-- Transform project name for RootNamespace: replace spaces and dashes with underscores, prefix with underscore if starts with digit -->
+    <RootNamespace>$(MSBuildProjectName.Replace(" ", "_").Replace("-", "_"))</RootNamespace>
+    <RootNamespace Condition=" '$([System.Char]::IsDigit($(RootNamespace), 0))' == 'true' ">_$(RootNamespace)</RootNamespace>
```

Comment:
nit: you don't need the string-wrapping or comparison to `'true'` - you can just use the output of `IsDigit` itself